sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -877,6 +878,71 @@ case class GpuFromUnixTime(
   override lazy val resolved: Boolean = childrenResolved && checkInputDataTypes().isSuccess
 }
 
+
+class FromUTCTimestampExprMeta(
+    expr: FromUTCTimestamp,
+    override val conf: RapidsConf,
+    override val parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends BinaryExprMeta[FromUTCTimestamp](expr, conf, parent, rule) {
+
+  override def tagExprForGpu(): Unit = {
+    extractStringLit(expr.right) match {
+      case None =>
+        willNotWorkOnGpu("timezone input to GpuFromUTCTimestamp must be a literal string")
+      case Some(timezoneShortID) =>
+        if (timezoneShortID != null) {
+          val utc = ZoneId.of("UTC").normalized
+          // This is copied from Spark, to convert `(+|-)h:mm` into `(+|-)0h:mm`.
+          val timezone = ZoneId.of(timezoneShortID.replaceFirst("(\\+|\\-)(\\d):", "$10$2:"),
+            ZoneId.SHORT_IDS).normalized
+
+          if (timezone != utc) {
+            willNotWorkOnGpu("only timezones equivalent to UTC are supported")
+          }
+        }
+    }
+  }
+
+  override def convertToGpu(timestamp: Expression, timezone: Expression): GpuExpression =
+    GpuFromUTCTimestamp(timestamp, timezone)
+}
+
+case class GpuFromUTCTimestamp(timestamp: Expression, timezone: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def left: Expression = timestamp
+  override def right: Expression = timezone
+  override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType, StringType)
+  override def dataType: DataType = TimestampType
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    throw new IllegalStateException(
+      "Cannot have time zone given by a column vector in GpuFromUTCTimestamp")
+  }
+
+  override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector = {
+    throw new IllegalStateException(
+      "Cannot have time zone given by a column vector in GpuFromUTCTimestamp")
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    if (rhs.getBase.isValid) {
+      // Just a no-op.
+      lhs.getBase.incRefCount()
+    } else {
+      // All-null output column.
+      GpuColumnVector.fromNull(lhs.getRowCount.toInt, dataType).getBase
```

Comment:
We normally shouldn't directly return the vector within a GpuColumnVector since the latter is considered to be the owner of the underlying cudf column vector and this code breaks that.  I recommend avoiding creating the GpuColumnVector entirely, e.g.:
```suggestion
      GpuColumnVector.columnVectorFromNull(lhs.getRowCount.toInt, dataType)
```